scripts/run_backend_tests.py

```diff
@@ -659,6 +691,44 @@ def check_coverage(
     process = subprocess.run(
         cmd, capture_output=True, encoding='utf-8', env=env,
         check=False)
+
+    coverage_output_lines = process.stdout.split('\n')
+    # The `process.stdout` is a string containing the coverage report,
+    # structured as follows:
+    # process_stdout_structure =
+    # Name       Stmts Miss Branch BrPart Cover Missing.
+    # -------------------------------------------------.
+    # file1.py      10    0      2      0   100%   12, 14.
+    # file2.py      20    0      4      0    90%   32, 34, 36, 38.
+    # -------------------------------------------------
+    # TOTAL        30    0      6      0   95%.
+    incomplete_coverage_lines: List[str] = []
+    # Lines with incomplete coverage are appenede to incomplete_coverage_lines.
+    for line_number, line in enumerate(coverage_output_lines):
+
+        # The line having incomplete coverage, not a dashed line(-----),
+        # and not the heading line is appended to incomplete_coverage_lines.
+
+        line_missing_full_coverage = line and (
+            ' 100%' not in line and '-----' not in line and 'Name' not in line
+        )
+        if line_missing_full_coverage:
+            # If a line indicates incomplete coverage then Heading,
```

Comment:
I don't really understand this. I think that, for conceptual clarity, `incomplete_coverage_lines` should only contain the actual data, i.e. lines with files with a percentage that's not 100.

All the other formatting can be added when you print the results out.

Reply:
@seanlip Currently, the coverage report is generated within this function, and by performing the formatting at this stage, I avoid the need for additional code duplication and complexity elsewhere (because there one more looping will be needed ). So i added formatting also on the time of getting lines having coverage less than 100%.

Would it be beneficial to consider formatting the report output in terms of complexity?

Reply:
I think the problem is that I can't understand what you are doing from reading the code.

Variables should clearly describe what you are storing in them, and comments should be added as needed to explain. When you say incomplete_coverage_lines I take that to mean exactly the lines that are not completely covered and nothing else. If you are including other stuff then you should call it something else. If you have difficulty coming up with a suitable name then that probably indicates you should use more easy-to-understand concepts instead, even if that means adding more logic.

In general: be clear about what your variables store, name them accordingly, and write the implementation to store exactly the right thing in each variable based on its name.

Reply:
> I think the problem is that I can't understand what you are doing from reading the code.
> 
> Variables should clearly describe what you are storing in them, and comments should be added as needed to explain. When you say incomplete_coverage_lines I take that to mean exactly the lines that are not completely covered and nothing else. If you are including other stuff then you should call it something else. If you have difficulty coming up with a suitable name then that probably indicates you should use more easy-to-understand concepts instead, even if that means adding more logic.
> 
> In general: be clear about what your variables store, name them accordingly, and write the implementation to store exactly the right thing in each variable based on its name.

 @seanlip, I'm attempting to refactor the code to enhance its readability, making it comprehensible for everyone to understand its functionality.

Reply:
@seanlip I've introduced a new function for improved code clarity, but I'm concerned it might not be easy to understand. -
![image](https://github.com/oppia/oppia/assets/97088265/969e74a0-c5b8-4c49-882b-ac6f823bb200)